### PR TITLE
Split ValueTypeSystemArraycopyTests to src_lw5 and src_qtypes folders

### PR DESCRIPTION
**Pull request by theresa-m**
Modify tests to compile with lw5

Related https://github.com/eclipse-openj9/openj9/issues/18157

### CODE REVIEW

**Comment by theresa-m** [comment]
@a7ehuo can you review my changes to make sure they reflect your intentions for these tests?

**Comment by theresa-m** [comment]
If you look at the second commit instead of all changes you can see only the changes that were made to the tests.

File: test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java [hunk 1]
```diff
@@ -59,46 +59,26 @@ public static class SomeIdentityClass2 implements SomeInterface {
...
-	public primitive static class SomePrimitiveValueClass2 implements SomeInterface {
-		long val1;
-		double val2;
+		public implicit SomeValueClass2();
 
-		SomePrimitiveValueClass2(int i) {
+		SomeValueClass2(int i) {
 			this.val1 = (long)(i*5);
 			this.val2 = (double)(i*6);
 		}
```
Comment (a7ehuo) [inline test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java:84]: `SomeValueClass` is meant to test just value object class. `SomePrimitiveValueClass` is meant to test Null Restricted flattenable value object class. `SomePrimitiveValueClass` shouldn't be removed. I think `SomeValueClass2` shouldn't be primitive in the original version. The test could be updated below:

```c++
	public value static class SomeValueClass implements SomeInterface {
		double val1;
		long val2;
		SomeValueClass2 val3;
		int val4;

		SomeValueClass(int i) {
			this.val1 = (double)i;
			this.val2 = (long)(i*2);
			this.val3 = new SomeValueClass2(i*3);
			this.val4 = i*4;
		}
	}

	public value static class SomeValueClass2 implements SomeInterface {
		long val1;
		double val2;

		SomeValueClass2(int i) {
			this.val1 = (long)(i*5);
			this.val2 = (double)(i*6);
		}
	}

	public value static class SomeNullRestrictedValueClass implements SomeInterface {
		double val1;
		long val2;
		SomeNullRestrictedValueClass2! val3;
		int val4;

		public implicit SomeNullRestrictedValueClass();

		SomeNullRestrictedValueClass(int i) {
			this.val1 = (double)i;
			this.val2 = (long)(i*2);
			this.val3 = new (SomeNullRestrictedValueClass2!)(i*3);
			this.val4 = i*4;
		}
	}

	public value static class SomeNullRestrictedValueClass2 implements SomeInterface {
		long val1;
		double val2;

		public implicit SomeNullRestrictedValueClass2();

		SomeNullRestrictedValueClass2(int i) {
			this.val1 = (long)(i*5);
			this.val2 = (double)(i*6);
		}
	}
```

File: test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java [hunk 2]
```diff
@@ -110,21 +90,21 @@ public primitive static class SomePrimitiveValueClass2 implements SomeInterface
 	public static SomeIdentityClass[] idArraySrc = new SomeIdentityClass[ARRAY_SIZE];
 	public static SomeValueClass[] vtArrayDst = new SomeValueClass[ARRAY_SIZE];
 	public static SomeValueClass[] vtArraySrc = new SomeValueClass[ARRAY_SIZE];
-	public static SomePrimitiveValueClass[] primitiveVtArrayDst = new SomePrimitiveValueClass[ARRAY_SIZE];
-	public static SomePrimitiveValueClass[] primitiveVtArraySrc = new SomePrimitiveValueClass[ARRAY_SIZE];
+	public static SomeValueClass![] nullRestrictedVtArrayDst = new SomeValueClass![ARRAY_SIZE];
+	public static SomeValueClass![] nullRestrictedVtArraySrc = new SomeValueClass![ARRAY_SIZE];
```
Comment (a7ehuo) [inline test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java:94]: `SomeValueClass!`  => `SomeNullRestrictedValueClass!` or whatever new class name will be used based on the above comment. 

The same comment applies to the rest of the file as well.

File: test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java (hunk 1 above)
Comment (theresa-m) [reply test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java:84]: Thanks for the feedback. I think SomeValueClass could be used for regular value class and nullrestricted case but its probably easier to read this way.

Comment (theresa-m) [reply test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java]: I removed primitive from SomeValueClass2 in the qtypes version of the tests and updated the lw5 tests to use SomeNullRestrictedValueClass and SomeNullRestrictedValueClass2

**Review by a7ehuo — APPROVED** [review]
Thank you for making the update! LGTM

**Comment by hangshao0** [comment]
Jenkins test sanity,extended winval jdknext

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**